src/scc_hypervisor_collector/api/uploader.py

```diff
@@ -5,7 +5,59 @@
 collected from the specified backends to the SCC using the provided
 credentials.
 """
+import json
+import logging
+import gzip
+from typing import Dict
+from importlib_metadata import version
+import requests
+from requests.exceptions import RequestException
+from .configuration import SccCredsConfig
 
 
 class SCCUploader:
     """SCC Uploader for scc-hypervisor-collector."""
+
+    def __init__(self, scccredsconfig: SccCredsConfig):
+        """Initialiser for SCCUploader"""
+        self._log = logging.getLogger(__name__)
+
+        # save the parameters
+        self._scccredsconfig: SccCredsConfig = scccredsconfig
+        self.username = self._scccredsconfig.username
+        self.password = self._scccredsconfig.password
+
+        self.auth = requests.auth.HTTPBasicAuth(self.username, self.password)
+
+        self.headers = {
+            'X-Gatherer-Version':
+                version('virtual-host-gatherer'),
+            'X-Scc-Hypervisor-Collector-Version':
+                version('scc-hypervisor-collector')
+        }
+
+    def upload(self, hv_details: Dict) -> None:
+        """ Upload the collected details to SCC"""
+        base_url = 'https://scc.suse.com'
+        path = '/organizations/hypervisors'
```

Comment:
Similarly this could be parameterised in the __init__() method, or in this method, with this value as the default if not specified?

Such an approach makes it easy to override the default for testing purposes.